### ISSUE TITLE
Investigate bug in `test_cluster_error_logs`

### ISSUE THREAD

**Issue by javiersanchz** [wazuh/wazuh#26080]
|Wazuh version|
|---|
| 4.9.1-rc2 |

## Description

During the release of [Release 4.9.1 - Release Candidate 2 - Workload benchmarks metrics](https://github.com/wazuh/wazuh/issues/26050), the `test_cluster_error_logs` of `Reliability` failed due to the following error:

```console
Internal error processing request 'b'sendsync'': Error 3023 - Worker node is not connected to master
```
The specific error arises from the `sendsync` task, which cannot be performed due to the lack of connection to the `master`

- The generated artifacts can be found in this [comment](https://github.com/wazuh/wazuh/issues/26050#issuecomment-2385155681)

**Comment by javiersanchz** [comment]
## Update

- I was investigating the files generated by the launch environment and found the cause of the error.
- I still need to look into a few things to conclude the investigation and document everything.

**Comment by GGP1** [comment]
## Review

The error is caused by the fact that the worker's local server is started before the connection to the master node is established. This can be seen clearly in the logs of the worker 1

```console
# Node is restarted
2024/09/30 10:08:02 INFO: [Cluster] [Main] SIGNAL [(15)-(SIGTERM)] received. Exit...
2024/09/30 10:08:12 DEBUG: [Cluster] [Main] Removing '/var/ossec/queue/cluster/'.
2024/09/30 10:08:12 DEBUG: [Cluster] [Main] Removed '/var/ossec/queue/cluster/'.
# Local server is up and already handling requests sent by the local client
2024/09/30 10:08:13 INFO: [Local Server] [Main] Serving on /var/ossec/queue/cluster/c-internal.sock
2024/09/30 10:08:13 DEBUG: [Local Server] [Keep alive] Calculating.
2024/09/30 10:08:13 DEBUG: [Local Server] [Keep alive] Calculated.
2024/09/30 10:08:13 ERROR: [Local Server] [Main] Could not connect to master. Trying again in 10 seconds.
2024/09/30 10:08:23 ERROR: [Local Server] [Main] Could not connect to master. Trying again in 10 seconds.
2024/09/30 10:08:33 ERROR: [Local Server] [Main] Could not connect to master. Trying again in 10 seconds.
# Local client connected and sent a sendsync command
2024/09/30 10:08:41 DEBUG: [Local 359676] [Main] Connection received in local server.
# The worker hasn't connected to the master yet, so the requests fail
2024/09/30 10:08:41 ERROR: [Local 359676] [Main] Internal error processing request 'b'sendsync'': Error 3023 - Worker node is not connected to master
2024/09/30 10:08:41 DEBUG: [Local 359676] [Main] Disconnected 359676.
2024/09/30 10:08:42 DEBUG: [Local 1027908] [Main] Connection received in local server.
2024/09/30 10:08:42 ERROR: [Local 1027908] [Main] Internal error processing request 'b'sendsync'': Error 3023 - Worker node is not connected to master
2024/09/30 10:08:42 DEBUG: [Local 1027908] [Main] Disconnected 1027908.
2024/09/30 10:08:43 DEBUG: [Local 446676] [Main] Connection received in local server.
2024/09/30 10:08:43 ERROR: [Local 446676] [Main] Internal error processing request 'b'sendsync'': Error 3023 - Worker node is not connected to master
2024/09/30 10:08:43 DEBUG: [Local 446676] [Main] Disconnected 446676.
# The worker client connects to the master node server on port 1516
2024/09/30 10:08:43 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B736_manager_1] [Main] Successfully connected to master.
```

The master node server starts a few seconds earlier than when the worker connects, that's why there's almost a minute between the local server and the connection to the master.

```console
2024/09/30 10:08:38 INFO: [Master] [Main] Serving on ('0.0.0.0', 1516)
2024/09/30 10:08:38 DEBUG: [Master] [Keep alive] Calculating.
2024/09/30 10:08:38 DEBUG: [Master] [Keep alive] Calculated.
2024/09/30 10:08:38 INFO: [Master] [Local integrity] Starting.
2024/09/30 10:08:38 INFO: [Master] [Local agent-groups] Sleeping 30s before starting the agent-groups task, waiting for the workers connection.
2024/09/30 10:08:38 INFO: [Master] [Local integrity] Finished in 0.106s. Calculated metadata of 34 files.
2024/09/30 10:08:43 INFO: [Worker] [Main] Connection from ('172.31.12.223', 39956)
```

We could fix this issue by ensuring that the workers connect to the master node before their local server is started.

We should decide whether to implement this change or leave this as it is, since `sendsync` won't be part of the new cluster. I'll leave the issue on hold until we know how to proceed.

**Comment by javiersanchz** [comment]
## Update

- The investigation will continue to find out why this is happening, as it is also occurring in https://github.com/wazuh/wazuh/issues/26204.

**Comment by javiersanchz** [comment]
## Update

I am still investigating this log, as it is related to the termination of the entire cluster.

Within both artifacts, both `4.8.1` and `4.9.1`, we find events related to `SIGTERM` regarding the termination of tasks during the restart, such as these, which are expected, in the `ossec.log` of the masters:

```console
2024/09/30 10:08:03 wazuh-monitord: INFO: (1225): SIGNAL [(15)-(Terminated)] Received. Exit Cleaning...
2024/09/30 10:08:03 wazuh-logcollector: INFO: (1225): SIGNAL [(15)-(Terminated)] Received. Exit Cleaning...
2024/09/30 10:08:03 wazuh-remoted: INFO: (1225): SIGNAL [(15)-(Terminated)] Received. Exit Cleaning...
2024/09/30 10:08:03 wazuh-syscheckd: INFO: (1756): Shutdown received. Releasing resources.
2024/09/30 10:08:03 wazuh-syscheckd: INFO: (1225): SIGNAL [(15)-(Terminated)] Received. Exit Cleaning...
2024/09/30 10:08:04 wazuh-analysisd: INFO: (1225): SIGNAL [(15)-(Terminated)] Received. Exit Cleaning...
2024/09/30 10:08:04 wazuh-execd: INFO: (1314): Shutdown received. Deleting responses.
2024/09/30 10:08:04 wazuh-execd: INFO: (1225): SIGNAL [(15)-(Terminated)] Received. Exit Cleaning...
2024/09/30 10:08:04 wazuh-db: INFO: (1225): SIGNAL [(15)-(Terminated)] Received. Exit Cleaning...
2024/09/30 10:08:05 wazuh-db: INFO: Graceful process shutdown.
2024/09/30 10:08:05 wazuh-authd: INFO: (1225): SIGNAL [(15)-(Terminated)] Received. Exit Cleaning...
2024/09/30 10:08:05 wazuh-authd: INFO: Exiting...
```

But the log from `... [Cluster] [Main] SIGNAL [(15)-(SIGTERM)] received. Exit...` is only found in the `cluster.log` of `4.9.1` environments, which involves the complete termination of the cluster and generates the associated errors.

I am still searching for the origin of this, comparing both branches, but I still can't find anything relevant that could cause this, the log files don't show any more information about it either.
It could be a resource issue in the environment, but the strange thing is that these `...Internal error processing request...` errors have been found in both releases of Workload benchmark metrics in `4.9.1`.

**Comment by javiersanchz** [comment]
## Update

- The problem has been verified locally.
- This error occurs when the cluster shuts down and restarts, during this time, the worker connects to the local server before connecting to the master, causing the error message.
- This does not affect the operation of any nodes, the cluster, or any behavior of the environment.
- On the other hand, the `SendSync` task will be removed in version `5.0.0`.
- Therefore, even though this error is being generated, it does not affect functionality in any way nor does it pose any threat.

**Comment by GGP1** [comment]
## Review

As I mentioned in my [previous review](https://github.com/wazuh/wazuh/issues/26080#issuecomment-2392093722), we could fix this issue by ensuring that the workers connect to the master node before their local server is started.

However, given that the error does not have much impact and `sendsync` is removed in `5.0.0`, it is likely that the change won't be implemented. LGTM

**Comment by fdalmaup** [comment]
# Review

The fix should be developed for `4.10.x` if a higher impact on the cluster behavior is found.
LGTM.
